src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/validator/AppointmentValidator.kt

```diff
@@ -79,35 +92,79 @@ class AppointmentValidator {
     }
   }
 
-  private fun validateAttendanceAndBehaviourFieldsIfHistoricAppointment(
+  private fun validateFeedbackFieldsIfHistoricAppointment(
     appointmentTime: OffsetDateTime,
+    didSessionHappen: Boolean?,
     attended: Attended?,
-    notifyProbationPractitioner: Boolean?,
+    late: Boolean?,
+    lateReason: String?,
+    noAttendanceInformation: String?,
+    noSessionReasonType: NoSessionReasonType?,
+    noSessionReasonPopAcceptable: String?,
+    noSessionReasonPopUnacceptable: String?,
+    noSessionReasonLogistics: String?,
     sessionSummary: String?,
     sessionResponse: String?,
     sessionConcerns: String?,
+    notifyProbationPractitioner: Boolean?,
     errors: MutableList<FieldError>,
   ) {
     // if the appointment occurred today or on a date in the future, no attendance validation is required
     if (appointmentTime.isAfter(OffsetDateTime.now().with(LocalTime.MIN))) {
       return
     }
 
+    if (didSessionHappen == null) {
+      errors.add(FieldError(field = "attendanceFeedback.didSessionHappen", error = Code.CANNOT_BE_EMPTY))
+      return
+    }
+
     if (attended == null) {
-      errors.add(FieldError(field = "appointmentAttendance.attended", error = Code.CANNOT_BE_EMPTY))
+      errors.add(FieldError(field = "attendanceFeedback.attended", error = Code.CANNOT_BE_EMPTY))
       return
     }
 
-    when (attended) {
-      NO -> {
-        checkValueNotSupplied(notifyProbationPractitioner, "sessionFeedback.notifyProbationPractitioner", Code.INVALID_VALUE, errors)
-        checkValueNotSupplied(sessionSummary, "sessionFeedback.sessionSummary", Code.INVALID_VALUE, errors)
-        checkValueNotSupplied(sessionResponse, "sessionFeedback.sessionResponse", Code.INVALID_VALUE, errors)
+    if (didSessionHappen) {
+      if (attended != YES) {
+        errors.add(FieldError(field = "attendanceFeedback.attended", error = Code.INVALID_VALUE))
+        return
+      }
+      if (late == null) {
+        errors.add(FieldError(field = "sessionFeedback.late", error = Code.CANNOT_BE_EMPTY))
+        return
+      } else if (late) {
```

Comment:
Do we need this `else if(late)`. Can't we not set `else`

Reply:
Yes we need if (late) because the lateReason field cannot be empty only if late is true, if late is false the field isn't required.